### ISSUE TITLE
Memory leak on com.firebase.ui.auch.KickoffActivity

### ISSUE THREAD

**Issue by viyt** [firebase/FirebaseUI-Android#1101]
### Describe your environment

  * Android device:Nexus 6P
  * Android OS version: 6.0.0
  * Google Play Services version: 3.1.1
  * Firebase/Play Services SDK version: 11.8.0
  * FirebaseUI version: 3.1.3
  
### Describe the problem:
After invoking  AuthUI.getInstance().createSignInIntentBuilder() from Activity, LeakCanary reported a leak on the KickoffActivity as it can be seen in the leak canary's report screenshot.

![2018-01-09 22 18 01](https://user-images.githubusercontent.com/23559779/34741836-ab7bf5ea-f58c-11e7-8369-97ca13415be7.png)

#### Steps to reproduce:

  1. Invoke  AuthUI.getInstance().createSignInIntentBuilder() from Activity 
  2. Wait for LeakCanary to run and report the leak
    
#### Observed Results:

  Screen of the leak above
  
#### Relevant Code:
  ```
   private void signIn() {
        startActivityForResult(
                AuthUI.getInstance().createSignInIntentBuilder()
                        .setLogo(R.drawable.logo_sp_mob)
                        .setTheme(R.style.AuthUITheme)
                        .setAvailableProviders(getSelectedProviders())
                        .setIsSmartLockEnabled(false)
                        .setAllowNewEmailAccounts(true)
                        .build(),
                RC_SIGN_IN);
    }

    private List<IdpConfig> getSelectedProviders() {
        Bundle params = new Bundle();
        params.putString(AuthUI.EXTRA_DEFAULT_COUNTRY_CODE, "uk");
        List<IdpConfig> selectedProviders = new ArrayList<>();
        selectedProviders.add(new IdpConfig.Builder(AuthUI.GOOGLE_PROVIDER).build());
        selectedProviders.add(new IdpConfig.Builder(AuthUI.FACEBOOK_PROVIDER).build());
        selectedProviders.add(new IdpConfig.Builder(AuthUI.EMAIL_PROVIDER).build());
        selectedProviders.add(new IdpConfig.Builder(AuthUI.PHONE_VERIFICATION_PROVIDER)
                .setParams(params).build());
        return selectedProviders;
    }
  ```

**Comment by samtstern** [comment]
@viyt thanks for the report!  Sounds like a bug, we will investigate.

**Comment by yash59** [comment]
any alternate solution?

**Comment by samtstern** [comment]
This has been fixed and released in version `3.2.0`.
